src/test/integration/gecko-profile-to-display.test.js

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { render } from 'react-testing-library';
+
+import { processProfile } from '../../profile-logic/process-profile';
+import ProfileCallTreeView from '../../components/calltree/ProfileCallTreeView';
+
+import { storeWithProfile } from '../fixtures/stores';
+import { getBoundingBox } from '../fixtures/utils';
+import mockCanvasContext from '../fixtures/mocks/canvas-context';
+import { createGeckoProfile } from '../fixtures/profiles/gecko-profile';
+
+beforeEach(() => {
+  // Mock out the 2d canvas for the loupe view.
+  jest
+    .spyOn(HTMLCanvasElement.prototype, 'getContext')
+    .mockImplementation(() => mockCanvasContext());
+  // This makes the bounding box large enough so that we don't trigger
+  // VirtualList's virtualization. We assert this above.
+  jest
+    .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+    .mockImplementation(() => getBoundingBox(1000, 2000));
+});
+
+describe('integration/gecko profile end to end', () => {
+  it('can display a gecko profile', () => {
+    const geckoProfile = createGeckoProfile();
+    const processedProfile = processProfile(geckoProfile);
+    const store = storeWithProfile(processedProfile);
+    render(
+      <Provider store={store}>
+        <ProfileCallTreeView hideThreadActivityGraph={true} />
```

Comment:
I'd be more comfortable putting this in `src/test/components/ProfileCallTreeView.test.js`. What do you think? This feels like a component test to me. It's intent is to provide an end-to-end test, but still using the component infrastructure. I would think a real integration test would spin up a real browser, which would be a bit different.

Reply:
yep, I can do it !